pkgs/development/python-modules/btrfs/default.nix

```
@@ -1,22 +1,31 @@
 { lib
 , buildPythonPackage
-, fetchPypi
+, fetchFromGitHub
 }:
 
 buildPythonPackage rec {
   pname = "btrfs";
-  version = "11";
+  version = "12";
 
-  src = fetchPypi {
-    inherit pname version;
-    sha256 = "1w92sj47wy53ygz725xr613k32pk5khi0g9lrpp6img871241hrx";
+  src = fetchFromGitHub {
+    owner = "knorrie";
+    repo = "python-btrfs";
+    rev = "v${version}";
+    sha256 = "sha256-ZQSp+pbHABgBTrCwC2YsUUXAf/StP4ny7MEhBgCRqgE=";
   };
 
+  postInstall = ''
+    mkdir -p "$out/bin"
+    mv bin/btrfs-* "$out/bin"
+  '';
+
+  pythonImportsCheck = [ "btrfs" ];
+
   meta = with lib; {
     description = "Inspect btrfs filesystems";
     homepage = "https://github.com/knorrie/python-btrfs";
     license = licenses.lgpl3Plus;
-    platforms = platforms.linux;
-    maintainers = [ maintainers.evils ];
+    platforms = platforms.linux ++ platforms.darwin;
```

if `btrfs-space-calculator` is useful on darwin, any reason to think it's not useful on all unix systems?

```suggestion
    platforms = platforms.unix;
```

---

Some python packages don't have `platforms` at all. Is this equivalent to `platforms.unix` or something else?

---

i'm not sure what you're asking
yes, `platforms.unix` is equivalent to `platforms.unix`,
that contains more than linux and darwin

... time elapses ...

turns out buildPythonPackage defaults meta.platforms to that of the interpreter,
which seems to be `platforms.all` for python 2.7 and `linux ++ darwin` for 3.8

the content of the source bin directory all has a python3 shebangs,
so just removing the platforms line is equivalent to your linux ++ darwin line,
and stays correct if our python3 gets built for more platforms
(updating my suggestion to match)

---

> if btrfs-space-calculator is useful on darwin, any reason to think it's not useful on all unix systems?

This is literally the only script that works on non-linux systems. Lets just not go down this rabbit hole and follow the upstreams meta.

https://github.com/knorrie/python-btrfs/blob/113a635793fb565d77fd89ed5e7ecfb8ebfaefc5/setup.py#L35

---

i'm not sure
on the one hand, if you use it on darwin and are willing to maintain that aspect of the package, it could be fine
on the other hand, even if you conditionally add the other scripts, presumably the python module itself is still useless on darwin, this could cause confusion
running a python script is somewhat trivial, so the cost of not providing it on darwin is small

---

I decided to not change `platforms` for now.